BUILD.md

```diff
@@ -163,6 +163,17 @@ list some basic and more used commands as example.
     ./waf --targets tests/test_math
     ```
 
+* **Use clang instead of gcc**
```

Comment:
Clang is used automatically on MacOS - maybe you can make it clear that this is only for linux

Reply:
Will do, thanks.